src/core/config/openthread-core-config-check.h

```diff
@@ -141,10 +141,6 @@
 #error "OPENTHREAD_ENABLE_DIAG was replaced by OPENTHREAD_CONFIG_DIAG_ENABLE."
 #endif
 
-#ifdef OPENTHREAD_ENABLE_LEGACY
-#error "OPENTHREAD_ENABLE_LEGACY was replaced by OPENTHREAD_CONFIG_LEGACY_ENABLE."
-#endif
```

Comment:
Keep this?
Instead add a new check for `OPENTHREAD_CONFIG_LEGACY_ENABLE` to say that its removed?